backend/app/usecase/validator/alias.go

```diff
@@ -12,19 +15,29 @@ type CustomAlias struct {
 }
 
 // IsValid checks whether the given alias has valid format.
-func (c CustomAlias) IsValid(alias *string) bool {
+func (c CustomAlias) IsValid(alias *string) (bool, Violation) {
 	if alias == nil {
-		return true
+		return true, Valid
 	}
 
 	if *alias == "" {
-		return true
+		return true, Valid
 	}
 
 	if len(*alias) >= customAliasMaxLength {
-		return false
+		return false, AliasTooLong
+	}
+
+	if c.HasFragmentCharacter(*alias) {
+		return false, HasFragmentCharacter
 	}
-	return true
+
+	return true, Valid
+}
+
+// HasFragmentCharacter returns whether the alias contains the '#' character which starts fragment identifiers in URLs
+func (c CustomAlias) HasFragmentCharacter(alias string) bool {
```

Comment:
I think we can rename it to `hasFragmentCharacter` because we will be using this function only inside `CustomAlias` struct.

Reply:
Done